muffnn/mlp/base.py

```diff
@@ -146,15 +149,29 @@ def partial_fit(self, X, y, monitor=None, **kwargs):
             for epoch in range(self.n_epochs):
                 self._random_state.shuffle(indices)
                 for start_idx in range(0, n_examples, self.batch_size):
+                    self.zeros=tf.constant(0, dtype=tf.float32,shape=[self.batch_size,self.n_classes_])
                     batch_ind = indices[start_idx:start_idx + self.batch_size]
+                    yb = y[batch_ind]
+                    msk = np.isfinite(yb)
+                    yb[~msk] = -1
                     feed_dict = self._make_feed_dict(X[batch_ind],
-                                                     y[batch_ind])
-                    obj_val, _ = self._session.run(
-                        [self._obj_func, self._train_step],
+                                                     yb)
+
+                    vals = self._session.run(
+                        [self._obj_func, self._train_step_step, self._selections, self._yfinite, self.ce] + self._checks,
                         feed_dict=feed_dict)
+                    obj_val = vals[0]
+                    selections = vals[2]
+                    yfin = vals[3]
+                    ce = vals[4]
```

Comment:
All of the code above needs to be changed back to the original form. This is leftover code from our hacking on this.